lambda/apiprocessor/search.py

```diff
@@ -37,16 +48,30 @@ def search(request):
 
     output = request
 
+    # extract the dates that the user wants to query
+    if 'date:' in keyword:
+        date_range = re.findall(r'date:\'\[(.*)\]\'',keyword)[0]
```

Comment:
What do you think about making this regex more restrictive? So that only YYYY-MM-DD and -TO- is allowed inside the brackets?

Could we also have  a check to validate the date format ?

Reply:
Have you thought about cases where a user adds a search for multiple date range search?
date:[start TO end] AND DATE:[start to end] The current implementation  does not throw an error to the user but will fail silently in the backend.

Reply:
Good point. I will add the check and make sure it throws some error for invalid date formats. I will also make the regex more restrictive.